src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.util.ArrayList;
+
+public class Storage {
+    private ArrayList<Task> tempStorage;
+    private boolean isExit;
+    Storage(){
+        tempStorage = new ArrayList<>(100);
+        isExit = false;
+    }
+    public void addTaskToTemp(String toAdd){
+        tempStorage.add(new Task(toAdd));
+    }
+    public void addToDoToTemp(String toAdd){
+        tempStorage.add(new Todo(toAdd));
+    }
+    public void addDeadlineToTemp(String toAdd) throws DukeException{
+        if(!toAdd.contains(" /by ")){
+            throw new DukeException("No /by found in command.");
+        }
+        String description = toAdd.split(" /by ")[0];
+        if(toAdd.split(" /by ").length<2){
+            throw new DukeException("No event time found after keyword /by .");
+        }
+        String by = toAdd.split(" /by ")[1];
+        tempStorage.add(new Deadline(description,by));
+    }
+    public void addEventToTemp(String toAdd) throws DukeException{
```

Comment:
It is great, you have multiple Exception cases. the error message is meaningful to the user.